packages/core/src/types/Matrix2D.ts

```diff
@@ -0,0 +1,663 @@
+ *  tx, ty, 1]
+ *
+ * Note that because a Matrix2D ignores the z-values of each component vectors,
+ * it does not satisfy all properties of a "real" 3x3 matrix.
+ *
+ *   - A Matrix2D has no transpose
+ *   - A(B + C) = AB + AC does not hold for a Matrix2D
+ *   - (rA)^-1 = r^-1 A^-1, r != 0 does not hold for a Matrix2D
+ *   - r(AB) = (rA)B = A(rB) does not hold for a Matrix2D
+ */
+export class Matrix2D {
+  public readonly values: Float32Array = new Float32Array(6);
+
+  public static get identity(): Matrix2D {
+    return new Matrix2D(1, 0, 0, 1, 0, 0);
+  }
+
+  public static get zero(): Matrix2D {
+    return new Matrix2D(0, 0, 0, 0, 0, 0);
+  }
+
+  public static fromRotation(angle: number): Matrix2D {
+    const matrix = Matrix2D.identity;
+    matrix.rotation = angle;
+    return matrix;
+  }
+
+  public static fromTranslation(
+    translation: PossibleVector2<number>,
+  ): Matrix2D {
+    const matrix = Matrix2D.identity;
+    matrix.translation = translation;
+    return matrix;
+  }
+
+  public static fromScaling(scale: PossibleVector2<number>): Matrix2D {
+    const matrix = Matrix2D.identity;
+    matrix.scaling = scale;
+    return matrix;
+  }
+
+  public get scaleX(): number {
+    return this.values[0];
+  }
+
+  public set scaleX(value: number) {
+    this.values[0] = value;
+  }
+
+  public set scaleY(value: number) {
+    this.values[3] = value;
+  }
+
+  public get scaleY(): number {
+    return this.values[3];
+  }
+
+  public get skew(): Vector2 {
+    return new Vector2(this.values[2], this.values[1]);
+  }
+
+  public set skew(skew: PossibleVector2) {
+    const vec = new Vector2(skew);
+    this.values[1] = vec.y;
+    this.values[2] = vec.x;
+  }
+
+  public set skewX(value: number) {
+    this.values[2] = value;
+  }
+
+  public get skewX(): number {
+    return this.values[2];
+  }
+
+  public set skewY(value: number) {
+    this.values[1] = value;
+  }
+
+  public get skewY(): number {
+    return this.values[1];
+  }
+
+  public set translateX(value: number) {
+    this.values[4] = value;
+  }
+
+  public get translateX(): number {
+    return this.values[4];
+  }
+
+  public set translateY(value: number) {
+    this.values[5] = value;
+  }
+
+  public get translateY(): number {
+    return this.values[5];
+  }
+
+  public set rotation(angle: number) {
+    angle = (angle * Math.PI) / 180;
+    const c = Math.cos(angle);
+    const s = Math.sin(angle);
+
+    this.values[0] = c;
+    this.values[1] = s;
+    this.values[2] = -s;
+    this.values[3] = c;
+  }
+
+  public get rotation(): number {
+    return (Math.atan2(this.values[1], this.values[0]) * 180) / Math.PI;
+  }
+
+  public set translation(translation: PossibleVector2<number>) {
+    const vec = new Vector2(translation);
+    this.values[4] = vec.x;
+    this.values[5] = vec.y;
+  }
+
+  public get translation(): Vector2 {
+    return new Vector2(this.values[4], this.values[5]);
+  }
+
+  public set scaling(value: PossibleVector2<number>) {
```

Comment:
Yes, the idea behind the setters was that they _set_ the value to a specific value, essentially overriding it. To transform the existing matrix you would use the `matrix.scale()` method, instead. I don't feel super strongly about this, but

```
matrix.scaling = 2;
```

implies that I want this matrix to represent a scaling of 2, i.e. _set_ the scaling to 2, not scale the current matrix by 2.

Reply:
I see, but technically, couldn't you normalize the 0th and 1st columns and multiply that by the `value`? 
This would achieve the same thing and would still work for rotated matrices, I think

Reply:
I guess the general question is if we even want to keep the setters. Because there's no point in having them behave the same way the methods do.

Reply:
Oh, sorry I had a bit of a brain fart there. You're saying that this method currently wouldn't work for a matrix that's already rotated because it's ignoring the horizontal and vertical skew, resulting in a shear. You're absolutely right.

Yeah, normalizing and multiplying should work. Do you think the setters are useful to keep around?

Reply:
Alright, I've updated both the setters for `scale`, `scaleX`, `scaleY`, and `rotation` so that they set the absolute value of the transformation while keeping the other transformations proportinal.

I've decided to remove the `skew` getters and setters for now as they seemed a little awkward and not that useful overall. Let me know if you'd like them back or not.

Reply:
I'm also an idiot and left a `test.only` in there 🤦

Reply:
Fixed 👍